lambdas/tests/unit/services/session-service.test.ts

```diff
@@ -218,5 +218,51 @@ describe("session-service", () => {
 
             expect(output.length).toEqual(36);
         });
+
+        it("should save the session data with context to dynamo db", async () => {
+            const mockSessionRequestSummary = {
+                clientId: "test-jwt-client-id",
+                clientIpAddress: "test-client-ip-address",
+                clientSessionId: "test-journey-id",
+                persistentSessionId: "test-persistent-session-id",
+                redirectUri: "test-redirect-uri",
+                state: "test-state",
+                subject: "test-sub",
+                context: "test-context",
+            };
+
+            jest.spyOn(global.Date, "now").mockReturnValueOnce(1675382400000);
+            jest.spyOn(configService, "getSessionExpirationEpoch").mockReturnValue(1675382500000);
+            jest.spyOn(configService, "getConfigEntry").mockReturnValue("session-table-name");
+            const output = await sessionService.saveSession(mockSessionRequestSummary);
+            expect(mockDynamoDbClient.prototype.send).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    input: expect.objectContaining({
+                        TableName: "session-table-name",
+                    }),
+                }),
+            );
+            expect(mockDynamoDbClient.prototype.send).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    input: expect.objectContaining({
+                        Item: expect.objectContaining({
+                            attemptCount: 0,
+                            clientId: "test-jwt-client-id",
+                            clientIpAddress: "test-client-ip-address",
+                            clientSessionId: "test-journey-id",
+                            createdDate: 1675382400000,
+                            expiryDate: 1675382500000,
+                            persistentSessionId: "test-persistent-session-id",
+                            redirectUri: "test-redirect-uri",
+                            state: "test-state",
+                            subject: "test-sub",
+                            context: "test-context",
+                        }),
+                    }),
+                }),
+            );
+
+            expect(output.length).toEqual(36);
```

Comment:
Why is this significant? what does 36 mean?

Reply:
Output is the session id which is a random UUID. 36 is the length of a UUID. I guess the alternative would be to validate the UUID with regex? 

Although, yes, I did just copy this from the above test.

Reply:
I've pushed up a change to make it a regex check to be clearer